hashintel/hash
# [FE] Make EditableCell in table responsive

## 🌟 What is the purpose of this PR?




The existing demo for table block doesn't work well on mobile. All the editable cells on the table have complete vertical scroll with each letter appearing on a new line. This PR fixes this by allowing each item in a cell to take the maximum width it can without causing a wrap.

## ⚠️ Known issues


- Since we're using `textarea`s for editable cells, we can't have it auto-wrap the content present inside it, ie, `width: max-content` won't work. Hence, I'm relying on using `max-width`s (which textarea does listen to) as a way to have sensible default widths for both desktop and mobile.

## 🔍 What does this change?




Check GitHub comments below

## 🔗 Related links





- [Improve table block rendering on mobile](https://app.asana.com/0/1201668052739245/1201303965906632/f) 

## ❓ How to test this?



- Checkout the branch and start the dev version of table block locally.
- Check the responsiveness of the cells in various widths.

## 📹 Demo

![chrome_paZqMXGSvw](https://user-images.githubusercontent.com/22196279/151978871-ae5fccf5-16fb-4e50-a3f4-288f8e233690.gif)
![chrome_w37geNGBJz](https://user-images.githubusercontent.com/22196279/151978876-fda25ee7-9477-4195-8cec-7a6abf417b77.png)
